src/Features/Core/Portable/AddImport/References/Reference.cs

```diff
@@ -92,26 +92,28 @@ public override int GetHashCode()
                 return Hash.CombineValues(this.SearchResult.NameParts);
             }
 
-            protected void ReplaceNameNode(
-                ref SyntaxNode contextNode, ref Document document, CancellationToken cancellationToken)
+            protected async Task<(SyntaxNode, Document)> ReplaceNameNodeAsync(
+                SyntaxNode contextNode, Document document, CancellationToken cancellationToken)
             {
-                if (!this.SearchResult.DesiredNameDiffersFromSourceName())
+                if (this.SearchResult.DesiredNameDiffersFromSourceName())
                 {
-                    return;
-                }
+                    var identifier = SearchResult.NameNode.GetFirstToken();
+                    var generator = SyntaxGenerator.GetGenerator(document);
+                    var newIdentifier = generator.IdentifierName(SearchResult.DesiredName).GetFirstToken().WithTriviaFrom(identifier);
+                    var annotation = new SyntaxAnnotation();
+
+                    var root = contextNode.SyntaxTree.GetRoot(cancellationToken);
+                    root = root.ReplaceToken(identifier, newIdentifier.WithAdditionalAnnotations(annotation));
+                    document = document.WithSyntaxRoot(root);
 
-                var identifier = SearchResult.NameNode.GetFirstToken();
-                var generator = SyntaxGenerator.GetGenerator(document);
-                var newIdentifier = generator.IdentifierName(SearchResult.DesiredName).GetFirstToken().WithTriviaFrom(identifier);
-                var annotation = new SyntaxAnnotation();
+                    var currentRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
almost all this code is just an indent change, but here's the important line.  

In the previous code, we made the new document, but *wrongly* got the 'contextNode' off of the root *prior* to making the new document.  Making a new document makes a *new* root, and we have to look the context node back up in that root.